### PR TITLE
Feature/viewer paged hotspot

### PR DESCRIPTION
**Pull request by angelodel80**
Aggiornato comportamento per attributo xml:id in pb

### CODE REVIEW

File: app/src/dataHandler/search/searchDocument.service.js [hunk 1]
```diff
@@ -235,4 +252,12 @@ angular.module('evtviewer.dataHandler')
             noteElements[0].parentNode.removeChild(noteElements[0]);
          }
       };
+   
+      XmlDoc.prototype.removeAddElements = function (xmlDocDom) {
```
Comment (szenzaro) [inline app/src/dataHandler/search/searchDocument.service.js]: This can probably be abstracted into a single function for `remove<Whatever>Elements`

something like:
```
removeElements(xmlDoc, ...elementsName){
         elementsName.forEach((en) => 
              Array.from(xmlDocDom.getElementsByTagName(en)).forEach((e) => e.parentNode.removeChild(e));
}
```

File: app/src/imageViewer/imageViewer.provider.js [hunk 2]
```diff
@@ -16,6 +16,11 @@ angular.module('evtviewer.openseadragon')
                 return true;
             }
 
+            imageViewer.imgCoeff = function(){
+                console.log('CHIAMATA PROVIDER for parameter: ', config.imageNormalizationCoefficient);
```
Comment (szenzaro) [inline app/src/imageViewer/imageViewer.provider.js]: console.log and italian comment...

File: app/src/imageViewer/imageViewer.service.js [hunk 3]
```diff
@@ -5,16 +5,18 @@
         var viewerModel = this;
         
         console.log("caricato servizio  imageViewerService");
-        console.log("#!!# con parsedData #!!#", parsedData);
-        console.log("#!!# page di parsedData #!!#", parsedData.getPage('page1').source);
+
+        //var tileSources = [ "data/tails/sanmatteo.dzi", "data/tails/schema.dzi"];
```
Comment (szenzaro) [inline app/src/imageViewer/imageViewer.service.js]: remove commented code

File: app/src/imageViewer/imageViewer.service.js [hunk 4]
```diff
@@ -5,16 +5,18 @@
...
     
         var options = {
             id: "osd-img",
             prefixUrl: "bower_components/openseadragon/built-openseadragon/openseadragon/images/",
+            //tileSources: tileSources, // immagine SanMatteo
```
Comment (szenzaro) [inline app/src/imageViewer/imageViewer.service.js]: remove commented code

File: app/src/imageViewer/imageViewer.service.js [hunk 5]
```diff
@@ -5,16 +5,18 @@
...
             //data/test-img/vb/VB_fol_104v_big.jpg
             tileSources:[] , // immagine per rotulo Alba
             //tileSources: "data/test-img/quamusdzi/QuamusImgManuscript.dzi", // immagine per Marocco
             //overlays: [{
+            //rotationMode: OpenSeadragon.OverlayRotationMode.NO_ROTATION,
```
Comment (szenzaro) [inline app/src/imageViewer/imageViewer.service.js]: remove commented code

File: app/src/imageViewer/imageViewer.service.js [hunk 6]
```diff
@@ -100,8 +91,5 @@
     
     
     });
-    console.log("caricato modulo evtviewer.imageViewerService");
-})();
-/*
-*/
-
+    console.log("prova caricato modulo evtviewer.imageViewerService");
```
Comment (szenzaro) [inline app/src/imageViewer/imageViewer.service.js]: remove console.log

File: app/src/imageViewer/imageViewer.service.js [hunk 7]
```diff
@@ -51,15 +41,16 @@
...
-            //navigatorHeight:"95%",
-            //navigatorWidth:"6%"
+            navigatorLeft: "94%",
+            navigatorHeight: "100%",
+            navigatorWidth: "50%",
+            navigatorTop:"1%",
+
+
          };
+
```
Comment (szenzaro) [inline app/src/imageViewer/imageViewer.service.js]: ???

File: app/src/imageViewer/imageViewer.service.js [hunk 8]
```diff
@@ -51,15 +41,16 @@
...
-            //navigatorWidth: "6%",
-            //navigatorTop:"1%",
-            //navigatorLeft:"94%",
-            //navigatorHeight:"95%",
-            //navigatorWidth:"6%"
+            navigatorLeft: "94%",
+            navigatorHeight: "100%",
+            navigatorWidth: "50%",
+            navigatorTop:"1%",
+
```
Comment (szenzaro) [inline app/src/imageViewer/imageViewer.service.js]: ???

File: app/src/imageViewer/scrollmap.service.js [hunk 9]
```diff
@@ -3,85 +3,19 @@ angular.module('evtviewer.openseadragonService')
    .service('imageScrollMap', function (evtInterface) {
       console.log('in service imageScrollMap');
       var key = 'yPage';
-
       var imageScrollMap = this;
-      // DATI DI PROVA PER IL ROTULO DI SAN TEOBALDO
+      /* DATI PER SCROLL MAPPA SAN MATTEO
```
Comment (szenzaro) [inline app/src/imageViewer/scrollmap.service.js]: italian comment and project specific

File: app/src/imageViewer/scrollmap.service.js [hunk 10]
```diff
@@ -103,7 +39,7 @@ angular.module('evtviewer.openseadragonService')
                if (box.y < map[key + '1'].to) {
                   return (key + '1').substr(1).toLowerCase();
                } else {
-                  for (var i = 2; i <= map.size; i++) {
+                  for (var i = 1; i <= map.size; i++) {
```
Comment (szenzaro) [inline app/src/imageViewer/scrollmap.service.js]: why this change?

File: app/src/imageViewer/viewerOSD.directive.js [hunk 11]
```diff
@@ -56,11 +56,15 @@
 
                scope.osd = viewer;
                scope.$parent[attrs.name] = viewer;
+               var coeff = osd.imgCoeff();
+               console.log("coefficiente", coeff);
```
Comment (szenzaro) [inline app/src/imageViewer/viewerOSD.directive.js]: console.log and italian comment

File: app/src/imageViewer/viewerOSD.service.js [hunk 12]
```diff
@@ -2,16 +2,25 @@
    console.log('caricato modulo openseadragonService');
    angular.module('evtviewer.openseadragonService', ['evtviewer.interface'])
 
-      .service('imageViewerHandler', function (evtInterface, imageScrollMap, overlayHandler) {
-         const ImageNormalizationCoefficient = 1265; // per il rotulo 3500;
+      .service('imageViewerHandler', function (evtInterface, imageScrollMap, overlayHandler, parsedData) {
+         var ImageNormalizationCoefficient = undefined; // per il rotulo 3500;
+         //const ImageNormalizationCoefficient = 1265; // per il rotulo 3500;
```
Comment (szenzaro) [inline app/src/imageViewer/viewerOSD.service.js]: remove commented code

File: app/src/imageViewer/viewerOSD.service.js [hunk 13]
```diff
@@ -2,16 +2,25 @@
...
          const YminPan = 0.5;
          var divHotSpotToggle = false;
 
          var viewerHandler = this;
 
+         //viewerHandler.ImageNormalizationCoefficient = undefined;
```
Comment (szenzaro) [inline app/src/imageViewer/viewerOSD.service.js]: remove commented code

File: app/src/imageViewer/viewerOSD.service.js [hunk 14]
```diff
@@ -2,16 +2,25 @@
...
          viewerHandler.viewer = undefined;
          viewerHandler.scope = undefined;
 
+
+         viewerHandler.setImgCoeff = function (coeff) {
+            //viewerHandler.ImageNormalizationCoefficient = coeff;
```
Comment (szenzaro) [inline app/src/imageViewer/viewerOSD.service.js]: remove commented code

File: app/src/imageViewer/viewerOSD.service.js [hunk 15]
```diff
@@ -149,9 +182,25 @@
 
          viewerHandler.updateViewerPage = function (page) {
             console.log('updateViewerPages: ', viewerHandler.viewer, page);
-            viewerHandler.viewer.goToPage(page);
-           
-
+            var pageValue = page;
```
Comment (szenzaro) [inline app/src/imageViewer/viewerOSD.service.js]: Duplicate code of [line 63](https://github.com/evt-project/evt-viewer/pull/19/files#diff-ff06ab2d2e33fc979e86c9045841e5dbR63).

Better abstract to common function

File: app/src/imageViewer/viewerOSD.service.js [hunk 16]
```diff
@@ -59,6 +89,8 @@
             viewerHandler.viewer.viewport.fitBounds(newBounds, true);
          };
 
+         /* Trying to fix Home bug*/
```
Comment (szenzaro) [inline app/src/imageViewer/viewerOSD.service.js]: ????

File: app/src/imageViewer/viewerOSD.service.js [hunk 17]
```diff
@@ -125,11 +157,12 @@
...
                      }
                   }
+                  //aggiungere scroll con coordinate asse x - FS
                }
 
                //event.stopBubbling = true;
-            } catch (err) {
-               console.log('error in pan', err);
+            }} catch (err) {
+               //console.log('error in pan', err);
```
Comment (szenzaro) [inline app/src/imageViewer/viewerOSD.service.js]: empty catch?

File: app/src/dataHandler/search/diplomaticLbHandler.service.js [hunk 18]
```diff
@@ -12,7 +12,8 @@ angular.module('evtviewer.dataHandler')
             parId = 1,
             line = {},
             lineId,
-            currentLineNodes = {},
+            currentLineNodes = [],
```
Comment (szenzaro) [inline app/src/dataHandler/search/diplomaticLbHandler.service.js]: Why it changes type?

File: app/config/config-sanmatteo.json [hunk 19]
```diff
@@ -0,0 +1,180 @@
+{
+    "indexTitle": "Edizione digitale",
+    "webSite": "",
+    "logoUrl": "favicon.ico",
+    "dataUrl":"data/bellini/encodings/LL1_16.xml",
```
[the remark this answers is not in the document]
Comment (angelodel80) [reply app/config/config-sanmatteo.json]: This file has surely not to be part of the merge!
